ClassNotFoundException: dotty.tools.dottydoc.Main since sbt 1.12.0

## steps

### `a1/src/main/scala/A.scala`

```scala
class A
```

### `build.sbt`

```scala
val common = Def.settings(
  scalaVersion := "3.3.7"
)

val a1 = project.settings(common)

common

enablePlugins(ScalaUnidocPlugin)
```

### `project/build.properties`

```properties
sbt.version=1.12.0
```

### `project/plugins.sbt`

```scala
addSbtPlugin("com.github.sbt" % "sbt-unidoc" % "0.6.0")
```

run `sbt unidoc`

## problem

```
[error] java.lang.RuntimeException: java.lang.ClassNotFoundException: dotty.tools.dottydoc.Main
[error] 	at xsbt.DottydocRunner.run(DottydocRunner.java:65)
[error] 	at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error] 	at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error] 	at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error] 	at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error] 	at sbtunidoc.Unidoc$.apply(Unidoc.scala:53)
[error] 	at sbtunidoc.BaseUnidocPlugin$.$anonfun$baseUnidocSettings$1(BaseUnidocPlugin.scala:38)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] 	at sbt.Execute.work(Execute.scala:292)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[error] Caused by: java.lang.ClassNotFoundException: dotty.tools.dottydoc.Main
[error] 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
[error] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
[error] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
[error] 	at java.base/java.lang.Class.forName0(Native Method)
[error] 	at java.base/java.lang.Class.forName(Class.java:423)
[error] 	at java.base/java.lang.Class.forName(Class.java:414)
[error] 	at xsbt.DottydocRunner.run(DottydocRunner.java:59)
[error] 	at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error] 	at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error] 	at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error] 	at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error] 	at sbtunidoc.Unidoc$.apply(Unidoc.scala:53)
[error] 	at sbtunidoc.BaseUnidocPlugin$.$anonfun$baseUnidocSettings$1(BaseUnidocPlugin.scala:38)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] 	at sbt.Execute.work(Execute.scala:292)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[error] (Scalaunidoc / doc) java.lang.ClassNotFoundException: dotty.tools.dottydoc.Main
```

## expectation

<!-- Describe what you think should've happened. -->

## notes

- https://github.com/scalate/scalate/pull/799

--- a/src/main/scala/sbtunidoc/BaseUnidocPlugin.scala
+++ b/src/main/scala/sbtunidoc/BaseUnidocPlugin.scala
@@ -37,7 +37,7 @@ object BaseUnidocPlugin extends AutoPlugin {
         (unidoc / sourcePositionMappers).value,
         fileConverter.value)
     },
-    unidoc / compilers := Def.uncached((sc / compilers).value),
+    unidoc / compilers := Def.uncached((sc / doc / compilers).value),
     unidoc / sources := Def.uncached((unidoc / unidocAllSources).value.flatten.sortBy { _.getAbsolutePath }),
     unidoc / scalacOptions := Def.uncached((sc / doc / scalacOptions).value),
     unidoc / javacOptions := Def.uncached((sc / doc / javacOptions).value),

